flyteplugins/go/tasks/pluginmachinery/tasklog/plugin.go

```diff
@@ -65,8 +69,9 @@ type Plugin interface {
 }
 
 type TemplateLogPlugin struct {
-	DisplayName   string                     `json:"displayName" pflag:",Display name for the generated log when displayed in the console."`
-	TemplateURIs  []TemplateURI              `json:"templateUris" pflag:",URI Templates for generating task log links."`
-	MessageFormat core.TaskLog_MessageFormat `json:"messageFormat" pflag:",Log Message Format."`
-	Scheme        TemplateScheme             `json:"scheme" pflag:",Templating scheme to use. Supported values are Pod and TaskExecution."`
+	DisplayName         string                     `json:"displayName" pflag:",Display name for the generated log when displayed in the console."`
+	TemplateURIs        []TemplateURI              `json:"templateUris" pflag:",URI Templates for generating task log links."`
+	DynamicTemplateURIs []DynamicTemplateURI       `json:"dynamicTemplateUris" pflag:",Dynamic URI Templates for generating task log links."`
```

Comment:
Why do these need to be different?

Reply:
A `DynamicTemplateURI` has a kind that we use to [match](https://github.com/flyteorg/flyte/pull/4774/files#diff-a7959cb7781e8c67a136b74e9051e5f64bf9e18eb236043de3fd4c35b9188407R241) with the values passed via task template. The use case is if we have multiple dynamic log links configured, but the task only enables a subset of them.